### ISSUE TITLE
Failure in f-string interpretation

### ISSUE THREAD

**Issue by ullix** [python/cpython#125086]
# Bug report

### Bug description:

Quotes inside a function should not be considered to be part of an f-string!

```python
# python 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] on Linux Mint 21.2

>>> info="abcdefg"
>>> print(f"TEST: {info.replace("\n", " ")}")
  File "<stdin>", line 1
    print(f"TEST: {info.replace("\n", " ")}")
                                  ^
SyntaxError: unexpected character after line continuation character
```

### CPython versions tested on:

3.10

### Operating systems tested on:

Linux

**Comment by blhsing** [comment]
This has been fixed with [PEP-701](https://docs.python.org/3.12/whatsnew/3.12.html#pep-701-syntactic-formalization-of-f-strings) since Python 3.12.

**Comment by skirpichev** [comment]
Yes, it's a documented misfeature of pre-3.12 releases, see
https://docs.python.org/3.10/reference/lexical_analysis.html#f-strings
and
https://docs.python.org/3.12/whatsnew/3.12.html#whatsnew312-pep701

This is not going to be fixed in 3.11.
